### PR TITLE
[sw/runtime] move exception enums to ibex.h

### LINKED ISSUE #7190: [sw/device] Separate interrupt/exception enum definitions from handler code
Currently the interrupt/exception enum that represents `mcause` values is part of the handler library: https://github.com/lowRISC/opentitan/blob/99e893798a52a2ecb91c1403e1cc0e9ccda6dc35/sw/device/lib/handler.h#L8-L17

To use it one also has to adopt  the handler library's interrupt handlers which isn't ideal. It would be better for these to live in a separate header file so that, for example, the mask ROM could use them. While we are at we should clean up the names to make them more descriptive.

### PR DESCRIPTION
**Pull request by timothytrippel**
This moves the exception enums from the OTTF to the the more generic
runtime library such that it may be shared across other regions of the
code base more easily.

This fixes #7190.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
@alphan @mundaym lmk if you prefer a different location for these, the issue (#7190) was somewhat outdated as these no longer lived in `handler.h` but in the OTTF code in `ottf_isrs.h`.

File: sw/device/lib/runtime/ibex.h [hunk 1]
```diff
@@ -18,10 +18,37 @@
...
+typedef enum ibex_exc_id {
+  kInstrMisaligned = 0,
+  kInstrAccessFault = 1,
+  kIllegalInstrFault = 2,
+  kBreakpoint = 3,
+  kLoadAccessFault = 5,
+  kStoreAccessFault = 7,
+  kUserECall = 8,
+  kMachineECall = 11,
+  kIdMax = 31
```
Comment (alphan) [inline sw/device/lib/runtime/ibex.h:39]: nit: `k<type_name><variant_name>`. You may want to consider a shorter type name, e.g. `ibex_exc_t` and `kIbexExcInstrAccessFault`.

File: sw/device/lib/runtime/ibex.h [hunk 2]
```diff
@@ -18,10 +18,37 @@
...
+} ibex_exc_id_t;
+
+/**
+ * An Ibex internal IRQ type.
+ *
+ * This enum is used to decode RISC-V internal IRQs generated by Ibex.
+ */
+typedef enum ibex_internal_irq_id {
```
Comment (alphan) [inline sw/device/lib/runtime/ibex.h:47]: nit: same here.

**Review by alphan — APPROVED** [review]
Thanks @timothytrippel, just some minor comments on names (how exciting, right?).

**Review by cfrantz — APPROVED** [review]

File: sw/device/lib/runtime/ibex.h (hunk 1 above)
Comment (timothytrippel) [reply sw/device/lib/runtime/ibex.h]: fixed.

File: sw/device/lib/runtime/ibex.h (hunk 2 above)
Comment (timothytrippel) [reply sw/device/lib/runtime/ibex.h]: fixed.

**Pull request merged**